xdp.go

```diff
@@ -380,6 +385,50 @@ func NewSocket(Ifindex int, QueueID int, options *SocketOptions) (xsk *Socket, e
 		return nil, fmt.Errorf("syscall.Bind SockaddrXDP failed: %v", err)
 	}
 
+	colle_name := "/home/mydir/bpf.o"
+	colle, err := ebpf.LoadCollection(colle_name);
+	if err != nil {
+		xsk.Close()
+		return nil, fmt.Errorf("load colle fail :%v", err)
+	}
+
+	if prog := colle.Programs["xdp_redirect_udp_func"]; prog != nil {
+		xsk.program = prog
+	} else {
+		xsk.Close()
+		fmt.Println("xdp prog not exist")
+	}
+
+	if err = netlink.LinkSetXdpFdWithFlags(link, xsk.program.FD(), int(DefaultXdpFlags)); err != nil {
```

Comment:
https://github.com/asavie/xdp/pull/3 removed code similar to this from the XDP socket creation so we could support multiple RX/TX queues -- it isn't necessary or desired for each rx-queue / xdp-socket to re-attach the same eBPF.